cosmos/client.go

```diff
@@ -94,3 +110,29 @@ func (c *Client) BuildAndBroadcastMsg(msg sdktypes.Msg, accName, accPassword str
 		return nil, errors.New("i/o timeout")
 	}
 }
+
+// QueryWithData performs a query to a Tendermint node with the provided path
+// and a data payload. It returns the result and height of the query upon success
+// or an error if the query fails.
+func (c *Client) QueryWithData(path string, data []byte) ([]byte, int64, error) {
+	return c.query(path, data)
+}
+
+// query performs a query to a Tendermint node with the provided store name
+// and path. It returns the result and height of the query upon success
+// or an error if the query fails. In addition, it will verify the returned
+// proof if TrustNode is disabled. If proof verification fails or the query
+// height is invalid, an error will be returned.
+func (c *Client) query(path string, key cmn.HexBytes) (res []byte, height int64, err error) {
```

Comment:
Kind of duplicated with `Query`, maybe we should have `Query` that uses `query`

Reply:
The `Query` is needed here because `NewAccountRetriver` expects [NodeQuerier](https://github.com/cosmos/cosmos-sdk/blob/master/x/auth/types/account_retriever.go#L11)

Check out https://github.com/cosmos/cosmos-sdk/blob/master/client/context/query.go#L41 for Query implementation. I left `query` because probably at some point we will need `QueryWitData` as well.